### PR TITLE
Set adjusted to UTC if UTC timezone (#1932)

### LINKED ISSUE #1932: unable to write parquet file with UTC timestamp
**Describe the bug**
I cannot figure out how to write a parquet file with a timestamp column that gets encoded as UTC. All my efforts produce files with naive timestamps and no UTC metadata.

**To Reproduce**

Consider this program: it writes a tiny parquet file to `/tmp/q.parquet`. But using both `pqrs` and `pandas/pyarrow` on the resulting file shows that there is no timezone present -- the metric_date column is a naive timestamp.

```rust
use std::sync::Arc;

use arrow::{
    array::{StringArray, TimestampMillisecondArray},
    datatypes::{DataType, Field, Schema, TimeUnit},
    record_batch::RecordBatch,
};
use parquet::{
    arrow::arrow_writer::ArrowWriter,
    file::properties::{WriterProperties, WriterVersion},
};

fn main() {
    //let tz = Some("UTC".to_owned());
    let tz = None;
    let fields = vec![
        Field::new(
            "metric_date",
            DataType::Timestamp(TimeUnit::Millisecond, tz.clone()),
            false,
        ),
        Field::new("my_id", DataType::Utf8, false),
    ];
    let schema = Arc::new(Schema::new(fields));

    let my_ids = Arc::new(StringArray::from(vec!["hi", "there"]));
    let dates = Arc::new(TimestampMillisecondArray::from_vec(
        vec![1234532523, 1234124],
        tz,
    ));
    let batch = RecordBatch::try_new(schema.clone(), vec![dates, my_ids]).unwrap();

    let f = std::fs::File::create("/tmp/q.parquet").unwrap();
    let props = WriterProperties::builder()
        .set_writer_version(WriterVersion::PARQUET_2_0)
        .build();

    let mut writer = ArrowWriter::try_new(f, schema, Some(props)).unwrap();
    writer.write(&batch).unwrap();
    writer.close().unwrap();
    println!("Hello, world!");
}

```

**Additional context**
Tested using arrow="16.0.0" and parquet="16.0.0".

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #1932

# Rationale for this change
 
Prior to https://github.com/apache/arrow-rs/pull/1682/files#diff-103f6c92559ee00d93eff806b411e0c8c8d249564fb3f8674da1da7693f86cb1L393 the writer would set `is_adjusted_to_u_t_c` to true if any timezone was present in the arrow datatype. This was incorrect as normalization to UTC would never actually occur.

The fix was to always set `is_adjusted_to_u_t_c` to be false, which whilst correct, is overly conservative and represents a regression for the case where the timezone actually is UTC.

# What changes are included in this PR?

If the source array is already normalized to UTC, set `is_adjusted_to_u_t_c` to be true

# Are there any user-facing changes?

Yes, not sure if counts as breaking though

### CODE REVIEW

File: parquet/src/arrow/schema.rs [hunk 1]
```diff
@@ -300,10 +300,15 @@ fn arrow_to_parquet_type(field: &Field) -> Result<Type> {
                 .with_repetition(repetition)
                 .build()
         }
-        DataType::Timestamp(time_unit, _) => {
+        DataType::Timestamp(time_unit, tz) => {
+            let is_utc = tz
```
Comment (tustvold) [inline parquet/src/arrow/schema.rs:304]: I initially played around with using chrono-tz, but this is a non-trivial additional dependency, and does not appear to be being actively maintained. I think this should be good enough, until such a time as we potentially add coerce types functionality as part of #1666

File: parquet/src/arrow/schema.rs [hunk 2]
```diff
@@ -300,10 +300,15 @@ fn arrow_to_parquet_type(field: &Field) -> Result<Type> {
...
+                .as_ref()
+                .map(|tz| tz == "UTC" || tz == "+00:00" || tz == "-00:00")
```
Comment (martin-g) [inline parquet/src/arrow/schema.rs:306]: Technically GMT is the timezone - https://www.timeanddate.com/time/gmt-utc-time.html
Should `GMT` be added in the checks too ?

Comment (tustvold) [reply parquet/src/arrow/schema.rs:306]: I think I would rather keep this small and simple, rather than cover all timezones that happen to be equivalent. Longer term this may be handled as part of #1938

**Comment by jorgecarleitao** [comment]
While applying this fix in arrow2, I broke some of our integration tests against pyarrow. Coming to the specs, when the tz string is set in Arrow, it means that

1. the values `i64` are time-aware
2. the values are stored in UTC
3. the tz string represents the offset/timezone that needs to be taken into account when using the `i64`

In this PR's notation, the values are by definition normalized to be UTC, it is the user of the logical type that needs to de-normalize them if they need to apply offsets (e.g. to represent).

AFAIK this matches with what "is adjusted to UTC" in parquet means:

> A TIMESTAMP with isAdjustedToUTC=true is defined as the number of milliseconds, microseconds or nanoseconds (depending on the unit parameter being MILLIS, MICROS or NANOS, respectively) elapsed since the Unix epoch, 1970-01-01 00:00:00 UTC. Each such value unambiguously identifies a single instant on the time-line.

In other words, I _think_ that the previous behavior was correct (but I may be wrong)
